qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -557,12 +557,16 @@ private ResolvedJavaTypeInfo validateObjectPart(String namespace, ObjectPart obj
 
 		JavaTypeInfoProvider javaTypeInfo = objectPart.resolveJavaType();
 		if (javaTypeInfo == null) {
-			String literalJavaType = LiteralSupport.getLiteralJavaType(objectPart.getPartName()); 
+			String literalJavaType = LiteralSupport.getLiteralJavaType(objectPart.getPartName());
+			if (objectPart.isOptional()) {

```

Comment:
Move this test before 

`String literalJavaType = LiteralSupport.getLiteralJavaType(objectPart.getPartName());`

because we don't need to check if it is a litteral type if object part is optional.